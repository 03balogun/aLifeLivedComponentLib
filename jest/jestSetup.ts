/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
