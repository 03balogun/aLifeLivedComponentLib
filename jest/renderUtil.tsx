import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
