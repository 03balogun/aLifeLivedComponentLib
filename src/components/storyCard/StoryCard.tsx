import React from 'react';

// UI
import { Box } from '../../themes/box/Box';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { Text } from '../../themes/text/Text';
import { Avatar } from '../avatar/Avatar';

// Types
import { Icon } from '../icon/Icon';

type StoryCardProps = {
  onPress: () => void;
  onBookmarkPress: () => void;
  onLikePress: () => void;
  hasUserLikedStory: boolean;
  hasUserBookmarkedStory: boolean;
  story: {
    id: string;
    title: string;
    likes: number;
    avatar: string;
    topic: string;
  };
};

export const StoryCard: React.FC<StoryCardProps> = ({
  story,
  onPress,
  onBookmarkPress,
  onLikePress,
  hasUserBookmarkedStory,
  hasUserLikedStory,
}) => {
  return (
    <SystemButton onPress={() => onPress()}>
      <Box
        flexDirection="row"
        alignContent="center"
        alignItems="center"
        alignSelf="center"
        justifyContent="space-between"
        paddingBottom="s"
        marginBottom="l"
        backgroundColor="white"
        marginTop="s"
        padding="s"
        borderRadius="medium"
        borderWidth={0.5}
        boxShadow="md"
        borderColor="borderColor"
      >
        <Box width="100%">
          <Box flexDirection="row" alignItems="center">
            <Avatar
              source={{ uri: story.avatar, priority: 'normal' }}
              width={40}
              height={40}
            />
            <Box width="80%" marginLeft="s">
              <Text variant="body" numberOfLines={1}>
                {story.title}
              </Text>
            </Box>
          </Box>

          <Box alignItems="center">
            <Box
              backgroundColor="secondary"
              alignSelf="flex-start"
              marginTop="s"
              padding="xs"
              borderRadius="medium"
              paddingLeft="s"
              paddingRight="s"
            >
              <Text
                variant="bodyXSmall"
                color="textContrastHigh"
                numberOfLines={1}
              >
                {story.topic}
              </Text>
            </Box>
          </Box>

          <Box
            borderBottomWidth={1}
            borderColor="backgroundContrastMid"
            marginTop="s"
            marginBottom="s"
          />

          <Box flexDirection="row" alignItems="center" marginLeft="s">
            <SystemButton flexDirection="row" width="20%" onPress={onLikePress}>
              <Icon
                icon={[hasUserLikedStory ? 'fas' : 'far', 'heart']}
                color="secondary"
                testID="story-card-heart-icon"
              />
              <Text variant="bodyXSmall" marginLeft="s">
                {story.likes}
              </Text>
            </SystemButton>

            <SystemButton onPress={onBookmarkPress}>
              <Icon
                icon={[hasUserBookmarkedStory ? 'fas' : 'far', 'bookmark']}
                marginLeft="m"
                color="primaryContrast"
                testID="story-card-bookmark-icon"
              />
            </SystemButton>
          </Box>
        </Box>
      </Box>
    </SystemButton>
  );
};
