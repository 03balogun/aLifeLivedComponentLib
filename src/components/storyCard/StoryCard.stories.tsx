import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { StoryCard } from './StoryCard';

const storyData = [
  {
    id: 'test-story-one',
    title: 'Test title',
    duration: '20:00',
    likes: 10,
  },
  {
    id: 'test-story-two',
    title: 'Test title',
    duration: '20:00',
    likes: 10,
  },
  {
    id: 'test-story-three',
    title:
      'This is a long title story, so we can see how it looks with a really long title! The more I type the I hope it mess up the story card design',
    duration: '20:00',
    likes: 10,
  },
];

storiesOf('Story card', module)
  .addDecorator(Story => Story())
  .add('Default story card', () => (
    <>
      {storyData.map(story => (
        <StoryCard
          key={story.id}
          story={story}
          hasUserBookmarkedStory={false}
          hasUserLikedStory={false}
          onPress={action('Story card pressed')}
          onBookmarkPress={action('Bookmark pressed')}
          onLikePress={action('Heart pressed')}
        />
      ))}
    </>
  ))
  .add('Liked and bookmarked story card', () => (
    <>
      {storyData.map(story => (
        <StoryCard
          key={story.id}
          story={story}
          hasUserBookmarkedStory={true}
          hasUserLikedStory={true}
          onPress={action('Story card pressed')}
          onBookmarkPress={action('Bookmark pressed')}
          onLikePress={action('Heart pressed')}
        />
      ))}
    </>
  ));
