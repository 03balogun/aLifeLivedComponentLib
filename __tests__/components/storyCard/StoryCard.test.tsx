import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { StoryCard } from '../../../src/components/storyCard/StoryCard';

const onStoryCardPress = jest.fn();
const onBookmarkPress = jest.fn();
const onLikePress = jest.fn();
const defaultTitle = 'Test Title';
const defaultTopic = 'Test topic';
const bookmarkTestId = 'story-card-bookmark-icon';
const heartTestId = 'story-card-heart-icon';

const setup = (
  hasUserBookmarkedStory = false,
  hasUserLikedStory = false,
  title = defaultTitle,
  avatar = 'https://picsum.photos/id/237/200/300'
) =>
  render(
    <StoryCard
      onPress={onStoryCardPress}
      onBookmarkPress={onBookmarkPress}
      onLikePress={onLikePress}
      hasUserBookmarkedStory={hasUserBookmarkedStory}
      hasUserLikedStory={hasUserLikedStory}
      story={{
        id: 'testId',
        title: title,
        likes: 15,
        avatar: avatar,
        topic: defaultTopic,
      }}
    />
  );

describe('components >> story card ', () => {
  beforeEach(
    () => (
      onStoryCardPress.mockClear(),
      onBookmarkPress.mockClear(),
      onLikePress.mockClear()
    )
  );

  test('Should find story title', () => {
    const { getByText } = setup();
    expect(getByText(defaultTitle)).toBeDefined();
  });

  test('Should find story topic', () => {
    const { getByText } = setup();
    expect(getByText(defaultTopic)).toBeDefined();
  });

  describe('Given user presses story card', () => {
    test('Should play audio', () => {
      const { getByText } = setup();
      fireEvent.press(getByText(defaultTitle));
      expect(onStoryCardPress).toBeCalledTimes(1);
    });
  });

  describe('Given user has NOT liked story', () => {
    test('Should like story', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId(heartTestId));
      expect(onLikePress).toBeCalledTimes(1);
    });
  });

  describe('Given user has NOT bookmarked story', () => {
    test('Should bookmark story', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId(bookmarkTestId));
      expect(onBookmarkPress).toBeCalledTimes(1);
    });
  });

  describe('Given user HAS liked story', () => {
    test('Should render story card with like', () => {
      const { toJSON } = setup(true, false);
      expect(toJSON()).toMatchSnapshot();
    });

    describe('Given user un likes story', () => {
      test('Should un like story', () => {
        const { getByTestId } = setup(true, false);
        fireEvent.press(getByTestId(heartTestId));
        expect(onLikePress).toBeCalledTimes(1);
      });
    });
  });

  describe('Given user HAS bookmarked story', () => {
    test('Should render story card with like', () => {
      const { toJSON } = setup(false, true);
      expect(toJSON()).toMatchSnapshot();
    });

    describe('Given user un bookmarks story', () => {
      test('Should un bookmark story', () => {
        const { getByTestId } = setup(true, false);
        fireEvent.press(getByTestId(bookmarkTestId));
        expect(onBookmarkPress).toBeCalledTimes(1);
      });
    });
  });
});
