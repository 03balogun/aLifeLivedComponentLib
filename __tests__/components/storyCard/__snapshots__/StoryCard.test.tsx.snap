// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> story card  Given user HAS bookmarked story Should render story card with like 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#2828281e",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginTop": 8,
          "padding": 8,
          "paddingBottom": 8,
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#282828",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "width": "100%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 96,
                  "height": 40,
                  "width": 40,
                },
              ],
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "normal",
                "uri": "https://picsum.photos/id/237/200/300",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "marginLeft": 8,
                "width": "80%",
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 18,
                },
              ]
            }
          >
            Test Title
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F08969",
                "borderRadius": 8,
                "display": "flex",
                "marginTop": 8,
                "padding": 2,
                "paddingLeft": 8,
                "paddingRight": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 14,
                },
              ]
            }
          >
            Test topic
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#CBCBCB",
              "display": "flex",
              "marginBottom": 8,
              "marginTop": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="#F08969"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            secondaryFill={null}
            secondaryOpacity={0.4}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {},
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            testID="story-card-heart-icon"
            vbHeight={512}
            vbWidth={512}
            width={16}
          >
            <RNSVGGroup
              fill={4293953897}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                fill={4293953897}
                fillOpacity={1}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 14,
                  "marginLeft": 8,
                },
              ]
            }
          >
            15
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="#282828"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            secondaryFill={null}
            secondaryOpacity={0.4}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginLeft": 16,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            testID="story-card-bookmark-icon"
            vbHeight={512}
            vbWidth={384}
            width={16}
          >
            <RNSVGGroup
              fill={4280821800}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
                fill={4280821800}
                fillOpacity={1}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`components >> story card  Given user HAS liked story Should render story card with like 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#2828281e",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginTop": 8,
          "padding": 8,
          "paddingBottom": 8,
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#282828",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "width": "100%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 96,
                  "height": 40,
                  "width": 40,
                },
              ],
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "normal",
                "uri": "https://picsum.photos/id/237/200/300",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "marginLeft": 8,
                "width": "80%",
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 18,
                },
              ]
            }
          >
            Test Title
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F08969",
                "borderRadius": 8,
                "display": "flex",
                "marginTop": 8,
                "padding": 2,
                "paddingLeft": 8,
                "paddingRight": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 14,
                },
              ]
            }
          >
            Test topic
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#CBCBCB",
              "display": "flex",
              "marginBottom": 8,
              "marginTop": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginLeft": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="#F08969"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            secondaryFill={null}
            secondaryOpacity={0.4}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {},
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            testID="story-card-heart-icon"
            vbHeight={512}
            vbWidth={512}
            width={16}
          >
            <RNSVGGroup
              fill={4293953897}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                fill={4293953897}
                fillOpacity={1}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Array [
                Object {
                  "color": "#282828",
                  "display": "flex",
                  "fontFamily": "SFUIDisplay-Regular",
                  "fontSize": 14,
                  "marginLeft": 8,
                },
              ]
            }
          >
            15
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="#282828"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            secondaryFill={null}
            secondaryOpacity={0.4}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginLeft": 16,
                },
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            testID="story-card-bookmark-icon"
            vbHeight={512}
            vbWidth={384}
            width={16}
          >
            <RNSVGGroup
              fill={4280821800}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
                fill={4280821800}
                fillOpacity={1}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
